lib/rubocop/cop/metrics/line_length.rb

```diff
@@ -40,6 +42,45 @@ def max
           cop_config['Max']
         end
 
+        def allow_heredoc?
+          allowed_heredoc
+        end
+
+        def allowed_heredoc
+          cop_config['AllowHeredoc']
+        end
+
+        def extract_heredocs(processed_source)
+          heredocs = []
+          iter = processed_source.tokens.each
+          loop do
+            token = iter.next
```

Comment:
@fphilipe the `parser` gem already creates a `Parser::Source::Range` object for each AST node which can tell you the beginning and ending lines/columns where the source code for that node appears. Why don't you just find the AST nodes for the heredocs, note the lines/columns where they begin and end, and use that to check whether the end of a long line falls within a heredoc or not?